### PR TITLE
Simplify several profile calls

### PR DESCRIPTION
**Pull request by daniellepintz**
## What does this PR do?
Remove several calls to `profiler.profile` in favor doing the profiling inside `trainer._call_ttp_hook`

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/optimizer.py [hunk 1]
```diff
@@ -149,6 +149,7 @@ def closure_dis():
                 with opt_dis.toggle_model(sync_grad=accumulated_grad_batches):
                     opt_dis.step(closure=closure_dis)
         """
+        # is separating optimizer_step_without_closure and optimizer_step_with_closure still needed?
```
Comment (daniellepintz) [inline pytorch_lightning/core/optimizer.py:152]: question to reviewers

File: pytorch_lightning/loops/epoch/evaluation_epoch_loop.py [hunk 2]
```diff
@@ -107,8 +107,7 @@ def advance(  # type: ignore[override]
             raise StopIteration
 
         if not data_fetcher.store_on_device:
-            with self.trainer.profiler.profile("evaluation_batch_to_device"):
-                batch = self.trainer.training_type_plugin.batch_to_device(batch, dataloader_idx=(dataloader_idx or 0))
```
Comment (ananthsub) [inline pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:111]: this is very interesting. now everything will show up as `batch_to_device`. this is positive for consistency with users since there's no "evaluation_batch_to_device" hook users implement. on the other hand, what happens if users implement these hooks differently based on the running stage the trainer is in?

we don't profile the `setup` or `teardown` hooks any differently, so i wonder if `batch_to_device` has just been an anomaly this whole time

Comment (four4fish) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:111]: @ananthsub how can they implement these hooks differently based on the running stage the trainer? ttk passed in when create trainer which will be shared across stages right?

Comment (ananthsub) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:111]: @four4fish the training type plugin calls the lightning module hook. the lightning module hook can be overridden by users like this:

```
def transfer_batch_to_device(self, batch, device, dataloader_idx):
    if self.trainer.training:
        .....
   elif self.trainer.testing:
        ....
```

File: pytorch_lightning/core/optimizer.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/core/optimizer.py:152]: This is only relevant for manual optimization. It can be helpful information if the manual closure is particularly expensive.

File: pytorch_lightning/loops/epoch/evaluation_epoch_loop.py (hunk 2 above)
Comment (carmocca) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:111]: In addition to checking the running stage, the performance could be different if each stage's data is very different. But I don't think it warrants a specific profile name

**Review by carmocca — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/loops/epoch/evaluation_epoch_loop.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/loops/epoch/evaluation_epoch_loop.py:111]: this can (should) be mentioned in the changelog (and the ones below too)

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/core/optimizer.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/core/optimizer.py:152]: remember to remove the line before resolving
